packages/nuxt/src/pages/runtime/router.options.ts

```diff
@@ -57,12 +57,13 @@ function _getHashElementScrollMarginTop (selector: string): number {
   return 0
 }
 
-function _isDifferentRoute (a: RouteLocationNormalized, b: RouteLocationNormalized): boolean {
-  const samePageComponent = a.matched[0] === b.matched[0]
+function _isDifferentRoute (from: RouteLocationNormalized, to: RouteLocationNormalized): boolean {
+  const samePageComponent = from.matched[from.matched.length - 1] === to.matched[to.matched.length - 1]
+
   if (!samePageComponent) {
     return true
   }
-  if (samePageComponent && JSON.stringify(a.params) !== JSON.stringify(b.params)) {
+  if (samePageComponent && JSON.stringify(from.params) !== JSON.stringify(to.params)) {
```

Comment:
That would make a lot of sense but not sure why both `from.meta` and `to.meta` are empty objects at this stage

Reply:
Do you see any downside by directly comparing `from.path` to `to.path` instead?

Reply:
There's also a valid question about whether we scroll to top on query parameter change - feels like maybe we would. 🤔 

So in that case comparing `fullPath` might be the thing to do.

Reply:
Thought about that but since this piece of code was already there (I just updated the variable name from `a` / `b` to `from` / `to`) I thought it was covering some edge case. 

Maybe we can use 'path' instead as `fullPath` would include query params as well and I feel like these are quite likely to be manipulated by any page which uses filters for example (`?keyword=houses`) and would automatically trigger the scrolling. In that case users would have to manually scroll to top, any thoughts?

Reply:
Ah, you're quite right. Let's leave it unchanged for now then; any fix there isn't really strictly related to this PR.